src/QuLDE.jl

```diff
@@ -46,12 +47,92 @@ struct QuLDEParam{W,L,Q, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixB
     end
 end
 
-function quldecircuit(blk::QuLDEParam, M::Matrix)
-    T = Int(log2(blk.k+1))
-    siz, = size(M)
-    nbit = Int(log2(siz))
+struct QuLDEnonUnitParam{W,L,Q,NL,Nbit, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}, LType <: GeneralMatrixBlock{NL}}
+    k::Int
```

Comment:
can you leave some comment on which is which for each member here? It is really hard to read.